### PR TITLE
Workarounds for Swift runtime memory corruption

### PR DESCRIPTION
**Pull request by jspahrsummers**
“Fixes” crashes [like these](https://gist.github.com/jspahrsummers/b77aeb5ff0eabe67feb7), as reported in https://github.com/Carthage/Carthage/pull/368#issuecomment-87943571.

I basically just addressed these with the brute force approach: run until we crash, and then find another way to express the code there. :confused: 

/cc @robrix @jckarter

### CODE REVIEW

**Comment by jspahrsummers** [comment]
Ugh, now the tests are crashing.

**Comment by robrix** [comment]
I’ve found curried function definitions are prone to runtime crashes. Are we using them fairly widely…?

**Comment by jspahrsummers** [comment]
@robrix Yeah, they're essential to the functionality of `|>`.

**Comment by jspahrsummers** [comment]
No matter what I fix here, the crash always shifts around. Definitely some sort of corruption issue, and I'm not sure how to work around it. :weary:

**Comment by robrix** [comment]
@jspahrsummers Right, but what I mean is, sometimes this crashes:

```swift
func foo(x: Int)(_ y: Int) -> Int
```

where this does not:

```swift
func foo(x: Int) -> Int -> Int
```

This seems particularly common when the function has type constraints.

**Comment by jspahrsummers** [comment]
Merging from `swift-development` fixes test crashes, but now [the Carthage crash](https://github.com/Carthage/Carthage/pull/368) is back. :finnadie:

**Comment by robrix** [comment]
Curried function syntax is said to be fixed in Xcode 6.4.

**Comment by jspahrsummers** [comment]
Finally had a chance to update this branch and re-test it on Xcode 6.3.1.

It seems to be :sparkles: now, but that's probably more due to @ikesyo's work on other crashes than anything I changed here. :smile:

**Comment by robrix** [comment]
:tada:

File: ReactiveCocoa/Swift/Bag.swift [hunk 1]
```diff
@@ -48,6 +48,16 @@ internal struct Bag<T>: SequenceType {
...
+
+		var index = next + 1
+		while index > 0 {
+			--index
+			if let value = elements[index] {
+				values.append(value)
+			}
+		}
+
+		return GeneratorOf(values.generate())
```
Comment (robrix) [inline ReactiveCocoa/Swift/Bag.swift]: This makes me so sad!

Comment (robrix) [inline ReactiveCocoa/Swift/Bag.swift]: To be clear: this seems like a very good solution. Just a heartbreaking necessity.

**Comment by robrix** [comment]
I’m :+1: on all of this.

**Pull request merged**